### PR TITLE
Remove IceUtil::Handle and IceUtil::Shared

### PR DESCRIPTION
**Pull request by pepone**
This PR remove the remaining IceUtil::Handle and IceUtil::Shared usages from the C++ code base

### CODE REVIEW

File: cpp/src/Ice/Thread.cpp [hunk 1]
```diff
@@ -160,7 +160,7 @@ WINAPI startHook(void* arg)
         //
         // See the comment in IceUtil::Thread::start() for details.
```
Comment (externl) [inline cpp/src/Ice/Thread.cpp:161]: Is this comment still relevant?

File: cpp/src/IceSSL/OpenSSLTransceiverI.cpp [hunk 2]
```diff
@@ -458,7 +400,7 @@ OpenSSL::TransceiverI::initialize(IceInternal::Buffer& readBuffer, IceInternal::
         out << "SSL summary for " << (_incoming ? "incoming" : "outgoing") << " connection\n";
 
         //
-        // The const_cast is necesary because Solaris still uses OpenSSL 0.9.7.
+        // The const_cast is necessary because Solaris still uses OpenSSL 0.9.7.
```
Comment (externl) [inline cpp/src/IceSSL/OpenSSLTransceiverI.cpp:403]: Should we open an issue to get rid of this const cast?

**Review by externl — APPROVED** [review]
LGTM!

File: cpp/src/IceSSL/OpenSSLTransceiverI.cpp (hunk 2 above)
Comment (bernardnormier) [reply cpp/src/IceSSL/OpenSSLTransceiverI.cpp:403]: We should remove all Solaris CC -related code!

File: cpp/src/Ice/BatchRequestQueue.cpp [hunk 3]
```diff
@@ -23,7 +21,7 @@ class BatchRequestI : public Ice::BatchRequest
 public:
 
     BatchRequestI(BatchRequestQueue& queue, const Ice::ObjectPrxPtr& proxy, const string& operation, int size) :
-        _queue(queue), _proxy(proxy), _operation(operation), _size(size)
+        _queue(queue),_proxy(proxy), _operation(operation), _size(size)
```
Comment (bernardnormier) [inline cpp/src/Ice/BatchRequestQueue.cpp:24]: missing space

**Review by bernardnormier — APPROVED** [review]
Looks good.

File: cpp/src/IceIAP/Transceiver.h [hunk 4]
```diff
@@ -18,7 +18,10 @@
 namespace IceObjC
 {
 
-class iAPTransceiver final : public IceInternal::Transceiver, public IceInternal::StreamNativeInfo
+class iAPTransceiver final :
+    public IceInternal::Transceiver,
+    public IceInternal::StreamNativeInfo,
+    public std::enable_shared_from_this<iAPTransceiver>
```
Comment (externl) [inline cpp/src/IceIAP/Transceiver.h:24]: Do we also need to call the `enable_shared_form_this` constructor in the `iAPTransceiver` constructor?

**Review by externl — APPROVED** [review]

File: cpp/src/IceIAP/Transceiver.h (hunk 4 above)
Comment (externl) [reply cpp/src/IceIAP/Transceiver.h:24]: nvm, I'm not sure why I thought it was necessary.

**Pull request merged**